lib/facter/mysql_server_id.rb

```diff
@@ -14,9 +14,12 @@ def mysql_id_get
 end
 
 Facter.add('mysql_server_id') do
+  confine kernel: :linux
```

Comment:
I'm not sure if this is supposed to work on FreeBSD. In that case you could confine to not Windows

Reply:
As per my comment [here](https://github.com/voxpupuli/puppet-prometheus/pull/630#issuecomment-1202506554), confining to Linux has nothing to do with the issue you're encountering (very old ruby on your Windows agents still try to load facts even if they don't run them).

Reply:
Actually this is a ruby 2.5 feature as explained in https://stackoverflow.com/a/51726301/11953415 

This bug started showing up on our CentOS7 systems after rubocop to introduce the incompatible syntax in an April commit. https://github.com/puppetlabs/puppetlabs-mysql/commit/8095cce0affc3b8359a129685cef2f036dd1ae87

This autocommit by rubocop was in response to this change in March 2022 in which rubocop targets ruby version 2.5 instead of 2.4 https://github.com/puppetlabs/puppetlabs-mysql/commit/4ebc431250d0d665dbaca0824be2377caa2c3479#diff-4f894049af3375c2bd4e608f546f8d4a0eed95464efcdea850993200db9fef5c 

Could we just target ruby 2.4 again as per version 12.0.2?

Reply:
Why Ruby 2.4? Which Puppet Agent/Ruby Version do you use? Puppet packages Ruby 2.5/Puppet 6 and Ruby2.7/Puppet 7 in their AIO packages.

Reply:
@mobalt Thanks for the correction.  I must have been testing something different when I decided it was a ruby 2.0 thing.  The not having anything to do with confining to Linux is still the case though.

Targeting ruby 2.4 vs 2.5 is probably best discussed in the pdk-templates repo.  Specifically see this PR. https://github.com/puppetlabs/pdk-templates/pull/466